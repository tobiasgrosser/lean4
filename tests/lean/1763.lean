axiom P : Prop → Prop

@[congr]
axiom P_congr (a b : Prop) (h : a ↔ b) : P a ↔ P b

theorem ex1 {p q : Prop} (h : p ↔ q) (h' : P q) : P p := by
  simp [h]
  assumption

#print ex1

theorem ex2 {p q : Prop} (h : p = q) (h' : P q) : P p := by
  simp [h]
  assumption

#print ex2
