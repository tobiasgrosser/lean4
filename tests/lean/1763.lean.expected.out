theorem ex1 : ∀ {p q : Prop}, (p ↔ q) → P q → P p :=
fun {p q} h h' => Eq.mpr (id (propext (P_congr p q h))) h'
theorem ex2 : ∀ {p q : Prop}, p = q → P q → P p :=
fun {p q} h h' => Eq.mpr (id (propext (P_congr p q (Iff.of_eq h)))) h'
